src/components/user-block/react/UserBlock.tsx

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+
+import classNames from 'classnames';
+
+import { Theme, Themes } from 'LumX/components';
+
+import { COMPONENT_PREFIX } from 'LumX/core/react/constants';
+
+import { ThumbnailVariants } from 'LumX';
+import { Thumbnail } from 'LumX/components/thumbnail/react/Thumbnail';
+import { IGenericProps, getRootClassName } from 'LumX/core/react/utils';
+import { handleBasicClasses } from 'LumX/core/utils';
+
+/**
+ * Authorized size values.
+ */
+enum Sizes {
+    s = 's',
+    m = 'm',
+    l = 'l',
+}
+type Size = Sizes;
+
+/**
+ * Authorized Orientations.
+ */
+enum Orientations {
+    horizontal = 'horizontal',
+    vertical = 'vertical',
+}
+type Orientation = Orientations;
+
+/////////////////////////////
+
+/**
+ * Defines the props of the component.
+ */
+interface IUserBlockProps extends IGenericProps {
+    /* Avatar image. */
+    avatar?: string;
+    /* Simple Action block. */
+    simpleAction?: React.ReactNode;
+    /* Multiple Actions block. */
+    multipleActions?: React.ReactNode;
+    /* Additionnal fields used to describe the use. */
+    fields?: string[];
+    /* User name. */
+    name?: string;
+    /* Orientation. */
+    orientation?: Orientation;
+    /* Size. */
+    size?: Size;
+    /* Theme. */
+    theme?: Theme;
+    /* Callback for the click event. */
+    onClick?(): void;
+    /* Callback for the mouseEnter event. */
+    onMouseEnter?(): void;
+    /* Callback for the mouseEnter event. */
+    onMouseLeave?(): void;
+}
+type UserBlockProps = IUserBlockProps;
+
+/////////////////////////////
+
+/**
+ * Define the types of the default props.
+ */
+interface IDefaultPropsType extends Partial<UserBlockProps> {}
+
+/////////////////////////////
+//                         //
+//    Public attributes    //
+//                         //
+/////////////////////////////
+
+/**
+ * The display name of the component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const COMPONENT_NAME: string = `${COMPONENT_PREFIX}UserBlock`;
+
+/**
+ * The default class name and classes prefix for this component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const CLASSNAME: string = getRootClassName(COMPONENT_NAME);
+
+/**
+ * The default value of props.
+ *
+ * @type {IDefaultPropsType}
+ * @constant
+ * @readonly
+ */
+const DEFAULT_PROPS: IDefaultPropsType = {
+    orientation: Orientations.horizontal,
+    size: Sizes.m,
+    theme: Themes.light,
+};
+/////////////////////////////
+
+/**
+ * Render a user information as a card if orientation is vertical or no action user info block if horizontal.
+ *
+ * @return {React.ReactElement} The component.
+ */
+const UserBlock: React.FC<UserBlockProps> = ({
```

Comment:
`IUserBlockProps` instead of `UserBlockProps` ?

Reply:
That makes sense, I wanted to be consistant with the existing `switch` / `Button` components where a `type` is created from the interface.
done